docs/template-context.md

```diff
@@ -0,0 +1,41 @@
+---
+title: "Template Context"
+weight: 80
+---
+
+While using the generator tool you may want dynamic values to be populated to your templates and thus be rendered in the output. The generator is able to achieve that using the **template context**.
+The **template context** allows you to access the contents of the [AsyncAPI document](asyncapi-document.md) and inject dynamic values to the template files passed to the asyncAPI CLI during the generation process. The render engine will then display these dynamically assigned values in the output.
+
+## Template context generation process
+1. The **Generator** receives **Template** and **params** as input.
```

Comment:
```suggestion
1. The **Generator** receives the **Template** and **params** as input.
```

Reply:
In the generation process, Lukasz said to capitalize all the words in diagram e.g Parser, Template Context so the reader can easily understand the diagram. Similar to pull request 812

Reply:
yes, as this content describes the process show on diagram,  so capitalization has to match.

but if proper capitalization is what @alequetzalli suggests, then just apply it and update diagram accordingly please @Florence-Njeri

Reply:
ok, let's keep it as is then in this doc 😄